src/services/invitation/index.ts

```diff
@@ -55,6 +119,109 @@ const plugin: FastifyPluginAsync = async (fastify, options) => {
     },
   );
 
+  fastify.register(fastifyMultipart, {
+    limits: {
+      // fieldNameSize: 0,             // Max field name size in bytes (Default: 100 bytes).
+      // fieldSize: 1000000,           // Max field value size in bytes (Default: 1MB).
+      fields: 0, // Max number of non-file fields (Default: Infinity).
+      // allow some fields for app data and app setting
+      fileSize: 1024 * 1024 * 1024, // For multipart forms, the max file size (Default: Infinity).
+      files: 1024 * 1024 * 1024 * 1, // Max number of file fields (Default: Infinity).
+      // headerPairs: 2000             // Max number of header key=>value pairs (Default: 2000 - same as node's http).
+    },
+  });
+
+  fastify.route<{ Querystring: IdParam; Body: any }>({
+    method: HttpMethod.POST,
+    url: '/invitations/:id/upload_csv',
+    preHandler: fastify.verifyAuthentication,
+    handler: async (request, reply) => {
+      const {
+        member,
+        query: { id: parentId },
+        log,
+      } = request;
+      const errors: Error[] = [];
+      // check rights
+      try {
+        if (!parentId) {
+          throw new Error('User has passed an item located at its root filesystem');
+        }
+        //check permission
+        const repositories = buildRepositories();
+        const item = await repositories.itemRepository.get(parentId);
+        await validatePermission(repositories, PermissionLevel.Write, member, item);
+        const file = await request.file();
+
+        if (!file || file.mimetype != 'text/csv') {
+          throw new Error('An CSV file was not uploaded');
+        }
+        const { service: itemService } = items;
+        const stream = file.file;
+
+        const { rows, header } = await getCSV(stream);
+        const hasGrpCol = header.includes('group_name');
+        let groupNameToItem = new Map<string, Item>();
+        if (hasGrpCol) {
+          groupNameToItem = await rows.reduce(
```

Comment:
Not sure this needs to be an async reduce...

Reply:
But I would like to wait for the creation of folders, before creating the invitations, could you give me some advice on how to do this part?